Sources/Logging/Logging.swift

```diff
@@ -166,6 +166,20 @@ extension Logger {
         }
     }
 
+    /// Convenience function for merging multiple metadata items into a Logger's existing metadata.
+    ///
+    /// - parameters:
+    ///    - metadata: The metadata to merge into a Logger's existing metadata.
+    ///
+    /// - note: Logging metadata behaves as a value that means a change to the logging metadata will only affect the
+    ///         very `Logger` it was changed on.
+    @inlinable
+    public mutating func mergeMetadata(_ metadata: Logger.Metadata) {
```

Comment:
Can we add the `uniquingKeysWith` parameter as well [à la Dictionary](https://developer.apple.com/documentation/swift/dictionary/merge(_:uniquingkeyswith:)-m2ub)?